deploy/template.yaml

```diff
@@ -187,7 +187,20 @@ Resources:
         ServerSideEncryptionConfiguration:
           - ServerSideEncryptionByDefault:
               SSEAlgorithm: AES256
-      BucketName: !Sub ${AWS::StackName}-test-results
+      BucketName:
+        Fn::Join:
+            - '-'
+            - - Ref: AWS::StackName
+              - test-results
+              - Fn::Select:
+                  - 4
+                  - Fn::Split:
+                      - '-'
+                      - Fn::Select:
+                          - 2
+                          - Fn::Split:
+                              - /
+                              - Ref: AWS::StackId
```

Comment:
Also, does the StackId change even on an update only? Instinctively, I would have thought that when a stack is created, it maintains the same StackId throughout its lifecycle (though I'm struggling to find confirmation of how it works in the docs!)

Reply:
The stacks are the promotion stacks, which remain the same. The non-prod and prod stacks have different IDs though

Reply:
Ah, sorry, I realise I just misunderstood the issue here.